substratest/assets.py

```diff
@@ -194,13 +194,15 @@ class DataSample(_Asset):
 class ObjectiveDataset(_InternalStruct):
     dataset_key: str
     data_sample_keys: typing.List[str]
+    metadata: typing.Dict
```

Comment:
The ObjectiveDataset isn't really an asset, it's a part of the Objective. It won't have metadata attached to it.